testsuite/forge/src/backend/k8s/fullnode.rs

```
@@ -0,0 +1,738 @@
+        selector: LabelSelector {
+            match_labels: Some(labels_map.clone()),
+            ..LabelSelector::default()
+        },
+        volume_claim_templates: Some(vec![data_volume]), // a PVC that is created directly by the StatefulSet, and owned by it
+        template: PodTemplateSpec {
+            metadata: Some(ObjectMeta {
+                labels: Some(labels_map),
+                ..ObjectMeta::default()
+            }),
+            spec: Some(PodSpec {
+                containers: vec![Container {
+                    image: Some(fullnode_image),
+                    command: Some(vec![
+                        "/usr/local/bin/aptos-node".to_string(),
+                        "-f".to_string(),
+                        format!("/opt/aptos/etc/{}", FULLNODE_CONFIG_MAP_KEY),
+                    ]),
+                    volume_mounts: Some(vec![
+                        VolumeMount {
+                            mount_path: APTOS_CONFIG_VOLUME_PATH.to_string(),
+                            name: APTOS_CONFIG_VOLUME_NAME.to_string(),
+                            ..VolumeMount::default()
+                        },
+                        VolumeMount {
+                            mount_path: APTOS_DATA_VOLUME_PATH.to_string(),
+                            name: APTOS_DATA_VOLUME_NAME.to_string(),
+                            ..VolumeMount::default()
+                        },
+                        VolumeMount {
+                            mount_path: GENESIS_CONFIG_VOLUME_PATH.to_string(),
+                            name: GENESIS_CONFIG_VOLUME_NAME.to_string(),
+                            ..VolumeMount::default()
+                        },
+                    ]),
+                    // specifically, inherit resources, env,ports, securityContext from the validator
+                    ..validator_stateful_set
+                        .spec
+                        .as_ref()
+                        .unwrap()
```

I dont think we should unwrap here, we can raise the error upwards, but unwrapping will panic in a cryptic way